source/fundamentals/crud/write-operations/upsert.txt

```diff
@@ -4,8 +4,200 @@ Insert or Update in a Single Operation
 
 .. default-domain:: mongodb
 
-If your application stores and modifies data in MongoDB, you probably use
-insert and update operations. In certain workflows, you may need to choose
-between an insert and update depending on whether the document exists.
-In these cases, you can streamline your application logic by using the
-``upsert`` option available in the following methods:
+Overview
+--------
+
+You can update documents by passing a query filter to the 
+:java-docs:`updateOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateOne(org.bson.conversions.Bson,java.util.List)>`, 
+:java-docs:`updateMany() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateMany(org.bson.conversions.Bson,java.util.List)>`
+or :java-docs:`replaceOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#replaceOne(org.bson.conversions.Bson,TDocument)>` methods
+
+Each method returns an ``AcknowledgedUpdateResult`` that tells us:
+
+- the number of documents that matched the query filter
+- the number of documents that got modified 
+- the id of the document that got upserted 
+
+An upsert is an update to a document or insert of a document.  An update
```

Comment:
nitpick:

This is a rough transition. It might be better to add an introduction to the concept before stating what an upsert is. Right now it reads "these methods accept a filter, and return some Thing. An upsert is another thing..."